### PR TITLE
Feature/sharing publicize

### PR DESCRIPTION
**Pull request by nbradbury**
This PR is the "Publicize" step of Android Sharing ([demo video](https://cloudup.com/cZ8bS6YipJo)). The UI - especially the detail view - isn't final, and the ability to share from/to multiple accounts isn't supported yet.

This PR was becoming too large already, so the final UI and feature set will come in a subsequent PR.

Note: Keep in mind that once you enable sharing, new posts will automatically be shared as soon as they're posted. To avoid embarrassment, I recommend disabling sharing after testing.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/publicize/PublicizeListActivity.java [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+
+        String tag = getString(R.string.fragment_tag_publicize_detail);
+        getFragmentManager().popBackStack(tag, FragmentManager.POP_BACK_STACK_INCLUSIVE);
+    }
+
+    private void showDetailFragment(int siteId,
+                                    PublicizeService service,
+                                    PublicizeConnection connection) {
+        if (isFinishing()) return;
+
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/publicize/PublicizeListActivity.java]: we should update the actionbar title when replacing fragment.

**Comment by maxme** [comment]
Can we reuse the same share icon as in Calypso (or same kind)?

**Comment by nbradbury** [comment]
> Can we reuse the same share icon as in Calypso (or same kind)?

I should've mentioned the current icon is just a placeholder until I have the actual one (which may come in a subsequent PR).

File: WordPress/src/main/java/org/wordpress/android/ui/publicize/PublicizeListActivity.java (hunk 1 above)
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/publicize/PublicizeListActivity.java]: > we should update the actionbar title when replacing fragment.

I don't understand - why is this necessary?

Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/publicize/PublicizeListActivity.java]: Not necessary, but it feels weird when you hit back from a detail view and have the same title in the action bar: https://cloudup.com/cABp1g5qoGg

**Comment by maxme** [comment]
Probably nothing we can do, but I can't login to my Tumblr account using the "Connect" button. The Webview spits some JS errors:

```
11-30 16:42:49.020 1741-1741/org.wordpress.android E/Web Console: Uncaught TypeError: Illegal constructor at https://secure.assets.tumblr.com/client/prod/app/context/loginandregister/index.js?_v=f0c4abf09acf9b3ed4ffb67439ef1be9:27
11-30 16:42:49.048 1741-1741/org.wordpress.android E/Web Console: Uncaught TypeError: Cannot read property 'loggingData' of undefined at https://secure.assets.tumblr.com/assets/scripts/dashboard.js?_v=91f606932fe0a7dc335207156e74c1a2:90
11-30 16:42:49.060 1741-1741/org.wordpress.android E/Web Console: Uncaught TypeError: Cannot call method 'on' of undefined at https://secure.assets.tumblr.com/assets/scripts/registration/registration.js?_v=3c00ad6c078e6f763a9942cb10781a1b:20
```

File: WordPress/src/main/java/org/wordpress/android/ui/publicize/PublicizeListActivity.java (hunk 1 above)
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/publicize/PublicizeListActivity.java]: > Not necessary, but it feels weird when you hit back from a detail view and have the same title in the action bar

Fair enough. I changed this in fc99fd3.

**Comment by nbradbury** [comment]
> Probably nothing we can do, but I can't login to my Tumblr account using the "Connect" button. The Webview spits some JS errors

Sadly, I think you're right - there's little we can do here :(

PS: Ready for another round.

**Comment by maxme** [comment]
I don't think that will make a big difference, but we could use an `IntentService` for PublicizeUpdateService. Probably less code, and everything will be run in a background thread.

**Comment by nbradbury** [comment]
> we could use an IntentService for PublicizeUpdateService

Good call. I made this change in 54c464a.

**Comment by maxme** [comment]
:shipit:
